router.go

```diff
@@ -122,6 +122,17 @@ func ParamsFromContext(ctx context.Context) Params {
 	return p
 }
 
+type matchKey struct{}
+
+// MatchedRouteKey is the request context key under which the handler path match is stored.
+var MatchedRouteKey = matchKey{}
+
+// MatchedRouteFromContext retrieves the matched route path from the context.
+func MatchedRouteFromContext(ctx context.Context) string {
```

Comment:
I'm not sure if it's even consistent in the existing code, but generally a *route* should be a tuple (Method, Path) in the context of this router. This is just the matched *path*.